src/context.rs

```diff
@@ -0,0 +1,321 @@
+use crate::{journal::ScrollJournal, ScrollSpecId};
+
+use derive_where::derive_where;
+use revm::{
+    context::{
+        Block, BlockEnv, Cfg, CfgEnv, ContextSetters, ContextTr, JournalTr, Transaction, TxEnv,
+    },
+    context_interface::context::ContextError,
+    database::EmptyDB,
+    Context, Database, MainContext,
+};
+
+/// Wraps a [`Context`] to correctly set the [`ScrollJournal`]'s spec.
+#[derive_where(Clone, Debug; BLOCK, CFG, CHAIN, TX, DB, <DB as Database>::Error)]
+pub struct ScrollContextFull<
```

Comment:
I order to correctly keep track of the `ScrollSpecId` in the `ScrollJournal`, we add a wrapper around the `Context`. This wrapper is only used to call `set_scroll_spec_id` whenever the spec id needs to be updated.